editor/src/components/inspector/controls/advanced-grid-modal.tsx

```diff
@@ -0,0 +1,251 @@
+import React from 'react'
+import { InspectorModal } from '../widgets/inspector-modal'
+import { FlexColumn, H1, SquareButton, UtopiaStyles } from '../../../uuiui'
+import { UIGridRow } from '../widgets/ui-grid-row'
+import type { OptionChainOption } from './option-chain-control'
+import { OptionChainControl } from './option-chain-control'
+import { PrettyLabel } from '../sections/layout-section/flex-container-subsection/flex-container-controls'
+import type { InspectorInfo } from '../common/property-path-hooks'
+import { useInspectorLayoutInfo } from '../common/property-path-hooks'
+import {
+  RadixSelect,
+  regularRadixSelectOption,
+  separatorRadixSelectOption,
+} from '../../../uuiui/radix-components'
+import { optionalMap } from '../../../core/shared/optional-utils'
+import { FlexJustifyContent } from 'utopia-api/core'
+
+export interface AdvancedGridModalProps {
+  id: string
+  testId: string
+  key: string
+  popupOpen?: boolean
+  openPopup?: (id: string) => void
+  closePopup?: () => void
+  style?: React.CSSProperties
+  pickerOffset?: {
+    x: number
+    y: number
+  }
+}
+
+export const AdvancedGridModal = React.memo((props: AdvancedGridModalProps) => {
+  const pickerOffset = props.pickerOffset ?? { x: -280, y: -20 }
+  const [dropdownOpen, setDropdownOpen] = React.useState({
+    justifyContent: false,
+    alignContent: false,
+  })
+  const toggleJustifyContentDropdown = React.useCallback((state: boolean) => {
+    setDropdownOpen((prev) => ({ ...prev, justifyContent: state }))
+  }, [])
+  const toggleAlignContentDropdown = React.useCallback((state: boolean) => {
+    setDropdownOpen((prev) => ({ ...prev, alignContent: state }))
+  }, [])
+  const closePopup = React.useCallback(() => {
+    if (!dropdownOpen.justifyContent && !dropdownOpen.alignContent) {
+      props.closePopup?.()
+    }
+  }, [dropdownOpen, props])
+
+  const alignItemsControl = useInspectorLayoutInfo('alignItems')
+  const justifyItemsControl = useInspectorLayoutInfo('justifyItems')
+  const alignContentControl = useInspectorLayoutInfo('alignContent')
+  const justifyContentControl = useInspectorLayoutInfo('justifyContent')
```

Comment:
I don't think we call the return value of `useInspectorLayoutInfo` "control" anywhere, we usually reserve the word Control for an actual UI element